module/VuFind/src/VuFind/Cover/Loader.php

```diff
@@ -697,6 +689,11 @@ protected function processImageURL($url, $cache = true)
             $tempFile = str_replace('.jpg', uniqid(), $this->localFile);
             $finalFile = $cache ? $this->localFile : $tempFile . '.jpg';
 
+            // Make sure that the cache directory exists.
+            if (!is_dir($directory = dirname($tempFile))) {
+                mkdir(dirname($tempFile), 0o755, true);
```

Comment:
There is an optional `dir_permission` setting in the [Cache] section of config.ini for directories in the cache. I wonder if we should use that here if defined, and default to `0o755` if it is not.

Reply:
It does appear that (apart from being refactored to a different place) the current code is essentially unchanged from @todolson's patch committed as https://github.com/vufind-org/vufind/commit/0719861390ea0ff54bb8d352d2beae410c590776.